language/move-lang/src/expansion/translate.rs

```diff
@@ -1427,21 +1434,53 @@ fn assign(context: &mut Context, sp!(loc, e_): P::Exp) -> Option<E::LValue> {
     use E::LValue_ as EL;
     use P::Exp_ as PE;
     let a_ = match e_ {
-        PE::Name(sp!(_, P::ModuleAccess_::ModuleAccess(..)), _)
-        | PE::Name(sp!(_, P::ModuleAccess_::QualifiedModuleAccess(..)), _)
-        | PE::Name(_, Some(_))
-            if !context.require_spec_context(
-                loc,
-                "only simple names allowed in assignment outside of specifications",
-            ) =>
+        PE::Name(n @ sp!(_, P::ModuleAccess_::ModuleAccess(_, _)), _)
+        | PE::Name(n @ sp!(_, P::ModuleAccess_::QualifiedModuleAccess(_, _)), _)
+            if !context.in_spec_context =>
         {
-            assert!(context.has_errors());
+            let msg = format!(
+                "Unexpected assignment of module access without fields outside of a spec \
+                 context.\nIf you are trying to unpack a struct, try adding fields, e.g. '{} {{}}'",
```

Comment:
There should be a comma after "e.g." -- also in the 2 other similar error messages below

Reply:
Style guidelines tend to vary on comma usage with i.e. and e.g.. Anything I've seen requires commas before, but as far as I've been able to tell, American and British style guidelines seem to split on a comma after--with the former requiring it and the latter not. 

I personally have always found commas after confusing (as it sets the i.e. and e.g. aside) and don't use them. I just think it reads better that way, and (I would imagine since I wrote most of them) it is the consistent choice with the rest of the error messages.